src/Workspaces/Core/Portable/Utilities/AbstractSpeculationAnalyzer.cs

```diff
@@ -381,14 +381,14 @@ private static bool SymbolsAreCompatibleCore(ISymbol symbol, ISymbol newSymbol,
                     var type = methodSymbol.ContainingType;
                     var newType = newMethodSymbol.ContainingType;
                     if ((type != null && newType != null) &&
-                        (
+                        ((
                             type.IsEnumType() &&
                             type.EnumUnderlyingType != null &&
                             type.EnumUnderlyingType.SpecialType == newType.SpecialType) ||
                         (
                             newType.IsEnumType() &&
                             newType.EnumUnderlyingType != null &&
-                            newType.EnumUnderlyingType.SpecialType == type.SpecialType))
+                            newType.EnumUnderlyingType.SpecialType == type.SpecialType)))
```

Comment:
the "newType" check looks exactly the same as the "type" check.   We should extract this into a helper method.  It will also make this actual complex expression much easier to read.